extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java

```diff
@@ -76,46 +79,46 @@ public String getUrl() throws ParsingException {
 
     @Override
     public String getName() throws ParsingException {
-        String name = null;
-        try {
-            name = videoInfo.getObject("title").getString("simpleText");
-        } catch (Exception ignored) {}
-        if (name == null) {
-            try {
-                name = videoInfo.getObject("title").getArray("runs").getObject(0).getString("text");
-            } catch (Exception ignored) {}
-        }
+        String name = getTextFromObject(videoInfo.getObject("title"));
         if (name != null && !name.isEmpty()) return name;
         throw new ParsingException("Could not get name");
     }
 
     @Override
     public long getDuration() throws ParsingException {
+        if (getStreamType() == StreamType.LIVE_STREAM) return -1;
+        String duration = null;
         try {
-            if (getStreamType() == StreamType.LIVE_STREAM) return -1;
-            return YoutubeParsingHelper.parseDurationString(videoInfo.getObject("lengthText").getString("simpleText"));
-        } catch (Exception e) {
-            throw new ParsingException("Could not get duration", e);
+            duration = getTextFromObject(videoInfo.getObject("lengthText"));
+        } catch (Exception ignored) {}
+        if (duration == null) {
+            try {
+                for (Object thumbnailOverlay : videoInfo.getArray("thumbnailOverlays")) {
+                    if (((JsonObject) thumbnailOverlay).getObject("thumbnailOverlayTimeStatusRenderer") != null) {
+                        duration = getTextFromObject(((JsonObject) thumbnailOverlay)
+                                .getObject("thumbnailOverlayTimeStatusRenderer").getObject("text"));
+                    }
+                }
+            } catch (Exception ignored) {}
         }
+        if (duration != null) return YoutubeParsingHelper.parseDurationString(duration);
+        throw new ParsingException("Could not get duration");
```

Comment:
This removes one check when the duration extractor was successful with first try
```suggestion
                }
            } catch (Exception ignored) {}
            
            if (duration == null) {
                throw new ParsingException("Could not get duration");
            }
        }
        return YoutubeParsingHelper.parseDurationString(duration);
```

Reply:
True, this probably happens at a lot of other places as well.